experimental/QuadFormAndIsom/src/embeddings.jl

```diff
@@ -213,6 +213,395 @@ function _overlattice(HAinD::TorQuadModuleMor,
+#   N as equiped with an isometry which we aim to extend. Note that for
+#   simplicity, if N has an isometry, M has one too; so we force M to be the one
+#   equiped with an isometry if only one lattice has an isometry to extend
+#   (since everything is symmetric);
+# - `even` forces the primitive extensions to be even - if all the data w.r.t to
+#   M, N and q does not allow for a primitive extension to be even, an error is
+#   thrown;
+# - `exist_only` is meant only to state about the existence of a primitive
+#   extensions without doing further computations once it is proved to exist;
+# - `first` asks to return the first primitive extensions comptued which
+#   satifies all the requirements;
+# - `fM` and `fN` are isometries of M and N we aim to extend. If no such
+#   isometries are specified, we set them to be the identity because we can always
+#   extend the identity along any extensions;
+# - `fqM` amd `fqN` are the representation of fM and fN on the respective
+#   discriminant groups. Since a priori fM and fN are seen as proper isometries of
+#   M and N, and fqM and fqN are constructed from an ambient isometry, we require
+#   to mention both;
+# - `x` is the index of the primitive extension (which is also the size of a
+#   glue kernel);
+# - `q` is the expected discriminant form of a primitive extension;
+# - `compute_bar_Gf`, in the equivariant cases (so ext_type != (:p, :p)) asks to
+#   compute the representation of the centralizer of the isometries constructed
+#   on the discriminant of the associated primitive extensions (can be computed
+#   at the level of glue map by "gluing stabilizers");
+# - `OqfM`: if `compute_bar_Gf == true`, then this is the image of the representation
+#   of the centralizer `O(M, fM)` on the discriminant group `qM`. It is needed
+#   to reconstruct the one of the extension (since a priori GM could be
+#   smaller);
+# - `OqfN`: same as before. Also, when the extension type of `(:e, :p)`, `OqfN`
+#   is the image of the representation of `O(N)` on the discriminant group `qN`
+#   used to reconstruct fitting isometries in the equivariant gluings;
+# - `discrep` in the case where M has an isometry but not N, N has to be
+#   definite and we want to find an isometry of N which coincide with the fixed
+#   isometry of M on the gluing. In that way, we can extend the isometry of M, and
+#   we classify all such isometries (this can be expensive)
+#
+#   TODO: add another argument to classify equivariant primitive extensions, but
+#   each such, consider only one isometry of N (it happens often that we do this
+#   classification up to conjugacy of a factor group of O(N), and thus at the end
+#   for each equivariant gluing, each extensions of fM give rises to the same
+#   cosets in the orthogonal group of the primitive extension).
+#
+# This generic function is then call by the different methods for primitive
+# extensions later
+function _primitive_extensions_generic(
```

Comment:
this function has a very long body. Would it make sense to split is up into sever functions?

Reply:
I will try.